### PR TITLE
feat: change entity search to use IQuery

### PR DESCRIPTION
**Pull request by dbouwman**
1. Description:

Change the entity search functions (`searchProjects, searchSites, HubProjectManager.search, HubSiteManager.search`) to take `IQuery` instead if `Filter<type>`

This also marks a lot of `/_internal` fns as `@private` b/c the doc generator does not check to see if fn's are exported from the package - just that the fn / type is "exported" at all.

None of these fns are used in hub-components or opendata-ui - so while techincally breaking change, it will not impact anything.

1. Instructions for testing:

- run tests

1. Closes Issues: [4392](https://devtopia.esri.com/dc/hub/issues/4392)

1. [x] used semantic commit messages
  
1. [x] PR title follows semantic commit format (**CRITICAL** if the title is not in a semantic format, the release automation will not run!)

### CODE REVIEW

File: packages/common/e2e/entity-search.e2e.ts [hunk 1]
```diff
@@ -18,51 +20,50 @@ describe("Entity Search", () => {
     jasmine.DEFAULT_TIMEOUT_INTERVAL = 200000;
   });
 
-  it("validate createEntitySearchFn", async () => {
+  fit("validate createEntitySearchFn", async () => {
```
Comment (tomwayson) [inline packages/common/e2e/entity-search.e2e.ts:23]:
```suggestion
  it("validate createEntitySearchFn", async () => {
```

File: packages/common/src/projects/HubProjects.ts [hunk 2]
```diff
@@ -246,27 +249,38 @@ export async function destroyProject(
...
  * Search for Projects, and get IHubProject results
  *
- * Different from `searchContent` in that this returns the specific entity type
+ * Different from `hubSearch` in that this returns the IHubProjects
  *
  * @param filter
  * @param options
  * @returns
  */
 export async function searchProjects(
```
Comment (tomwayson) [inline packages/common/src/projects/HubProjects.ts:258]: in the other managers, his method name is just `search()`, right? If so we could (and should) introduce that here, and we could leave and deprecate `searchProjects()` - one less breaking change.

File: packages/common/e2e/hub-sites.e2e.ts [hunk 3]
```diff
@@ -26,7 +26,7 @@ describe("Hub Sites", () => {
       // inspect and ensure we have what we expect
       expect(site.typeKeywords).toBeDefined();
       // get the slug keyword
-      const slug = site.typeKeywords.find((e) => e.indexOf("slug|") === 0);
+      const slug = site.typeKeywords?.find((e) => e.indexOf("slug|") === 0);
```
Comment (drspacemanphd) [inline packages/common/e2e/hub-sites.e2e.ts:29]: Does this automatically coerce to `[]`?

File: packages/common/src/sites/HubSites.ts [hunk 4]
```diff
@@ -489,18 +489,30 @@ export async function convertItemToSite(
...
-  };
-  // merge filters
-  const sitesFilter = mergeContentFilter([scopingFilter, filter]);
-  // delegate
-  return searchContentEntities(sitesFilter, convertItemToSite, options);
+): Promise<IHubSearchResponse<IHubSite>> {
+  let qry: IQuery;
+
+  if (typeof query === "string") {
+    qry = createQueryFromString(query, "term", "item");
```
Comment (drspacemanphd) [inline packages/common/src/sites/HubSites.ts:498]: Do we need the `term` argument here? As in, are we expecting any other filter to be applied as a string other than a general `term`?

File: packages/common/test/projects/HubProjects.test.ts [hunk 5]
```diff
@@ -71,18 +72,18 @@ const PROJECT_ITEM_ENRICH: portalModule.IItem = {
   ],
   description: "Test Project with data",
   tags: ["hubproject"],
-  snippet: null,
+  // snippet: null,
```
Comment (drspacemanphd) [inline packages/common/test/projects/HubProjects.test.ts:75]: why are these commented out?

**Review by tomwayson — APPROVED** [review]
one small change and a suggestion

**Review by drspacemanphd — APPROVED** [review]
Good work, I think the builds nicely in that we are trying to generalize search a bit more.

File: packages/common/src/projects/HubProjects.ts (hunk 2 above)
Comment (dbouwman) [reply packages/common/src/projects/HubProjects.ts:258]: In the Manager's the function is `search` b/c that's scoped into the Manager class... the backing module that just exports fn's needs to qualify the fn name - thus `searchProject`

Comment (tomwayson) [reply packages/common/src/projects/HubProjects.ts:258]: oops - I thought that was still part of the class, carry on.

File: packages/common/src/sites/HubSites.ts (hunk 4 above)
Comment (dbouwman) [reply packages/common/src/sites/HubSites.ts:498]: When searching for users, the generic thing may be `name`. I also considering enabling a query like `tag: water` which we could split and the use this same fn - or sniff for guid's and then use `id`. Viv built something like that in the early versions of the gallery and it was handy

File: packages/common/test/sites/HubSites.test.ts [hunk 6]
```diff
@@ -106,14 +107,14 @@ const SITE_ITEM_ENRICH: portalModule.IItem = {
     "Web Map",
     "Registered App",
   ],
-  description: null,
+  // description: null,
```
[the remark this answers is not in the document]
Comment (dbouwman) [reply packages/common/test/sites/HubSites.test.ts:110]: see above

**Pull request merged**